Controller/Methods/Cards.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace Omise\Payment\Controller\Methods;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Exception\NotFoundException;
+
+class Cards extends \Magento\Framework\App\Action\Action
+{
+    /**
+     * @var Session
+     */
+    protected $customerSession;
+
+    /**
+     * @var Magento\Framework\View\Result\PageFactory
+     */
+    private $pageFactory;
+
+    /**
+     * @param Context $context
+     * @param Session $customerSession
+     */
+    public function __construct(
+        Context $context,
+        Session $customerSession,
+        \Magento\Framework\View\Result\PageFactory $pageFactory
+    ) {
+        parent::__construct($context);
+        $this->customerSession = $customerSession;
+        $this->pageFactory = $pageFactory;
+    }
+
+    /**
+     * Dispatch request
+     *
+     * @param RequestInterface $request
+     * @return ResponseInterface
+     * @throws NotFoundException
+     */
+    public function dispatch(RequestInterface $request)
+    {
+        if (!$this->customerSession->authenticate()) {
+            $this->_actionFlag->set('', 'no-dispatch', true);
+        }
+        return parent::dispatch($request);
+    }
+
+    /**
+     * Dispatch request
+     *
+     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
+     * @throws \Magento\Framework\Exception\NotFoundException
+     */
+    public function execute()
+    {
+        $resultPage = $this->pageFactory->create();
+        $resultPage->getConfig()->getTitle()->set(__('Stored Payment Methods'));
```

Comment:
The table on the page only contains cards, and has a column heading reinforcing this fact. Shouldn't the page title be "Saved Payment Cards", "Saved Credit/Debit Cards", "Stored Payment Cards", or "Stored Credit/Debit Cards"?

Reply:
there will be other payment methods here in the future.

Reply:
BTW it is also exactly the same way of displaying as magento does it in `module_vault` `cards_list.phtml`

Reply:
So other payment methods will be in different tables on the same page? The table specifically only refers to cards. This contradiction should be resolved one way or the other

Reply:
@guzzilar Thoughts?

Reply:
@jacstn what would those other payment methods could possibly be (in the future)?

Reply:
don't know, but I have strong believes in omise.. any ideas about code?

Reply:
So I'm not sure what does `Stored Payment Methods` exactly mean?
About code, I think we should avoid over-design if possible unless we clearly see what is coming after and know what it is for. Or else it gives a confusion as for user experience.

Reply:
+1

Avoid putting unnecessary questions in the head of the user